### ISSUE TITLE
Separation of responsibility bug.

### ISSUE THREAD

**Issue by bogado** [gollum/gollum#1177]
The h1_title option of the wiki is help by the wiki object, that is defined on gollum-lib. The problem is that all, or most of, the processing is done in the application. This crossing of responsibilities is not good as the application is depending on the lib internals and the lib expects that the app works is a certain way.

My proposal is that the page field `@title` should reflect the options and the application will not need to know how the title was created. It would still initialize the wiki object passing the option, but it won't need to have logic inside it to reflect this.

I may have time to do that on my forks, but what I'm not sure is if there's other dependencies on the title reflecting the filename.

**Comment by bogado** [comment]
Pull requests with the change I was talking about : 

https://github.com/gollum/gollum/pull/1178  << on the application
https://github.com/gollum/gollum-lib/pull/233 << on the lib.

**Comment by dometto** [comment]
Thanks for this. What is the status of the PRs? I see failing tests, but I'm not sure if you're still working on this. I'm definitely in favor of separating the logic in the way you described, though.

**Comment by bogado** [comment]
I'll be working on this, but in a slow pace. I elected gollum to be the engine on my personal web site, that I could finally make it over into something rather than a nothing as it has been for a while.

The whole git/wiki idea is great and fits very well with my philosophy.

Both PRs are running on my site, there's limited testing done and the whole site is very, very, low volume. So there's a risk associated to them. 

During the time I studied the code I've detected some other places where the borders of the separation between the library and application could have been better marked. Some of the path/url handling for instance is suffering from this, I found out that is very easy to have a broken configurations that redirects to bad places as either the app or the lib don't have a clear picture of how it should be working.
